project/frontend/src/App.js

```diff
@@ -240,8 +96,21 @@ function App() {
             loggedIn={user.signedIn}
             innerNav={compId != 0}
             compIdChanged={setCompId}
-          ></HeaderBar>
-          <Layout>{pageRoute}</Layout>
+          />
+          {!authStateReceived ? (
+            <div>Signing in...</div>
+          ) : (
+            <Layout>
+              <PageRouter
+                signedIn={user.signedIn}
+                compId={compId}
+                loading={loading}
+                compIdChanged={setCompId}
+                competitionInfo={competitionInfo}
+                updateCompId={setCompId}
+              />
+            </Layout>
+          )}
```

Comment:
This is better, I'd still like all of the logic to be handled by `PageRouter` though, eg end state looks like:
```js
<div className="App">
  <AuthContext>
    <HeaderBar
      loggedIn={user.signedIn}
      innerNav={compId != 0}
      compIdChanged={setCompId}
    />
    <PageRouter
      authStateReceived={authStateReceived}
      signedIn={user.signedIn}
      compId={compId}
      loading={loading}
      compIdChanged={setCompId}
      competitionInfo={competitionInfo}
      updateCompId={setCompId}
    />
  </AuthContext>
</div>
```

Reply:
Refactored authStateReceived to be inside PageRouter as well, but I want to keep Layout on the outside, so that I don't have to wrap it around every return statement in PageRouter which would make it more verbose. Would this be ok? 
It now looks like:

```jsx
        <AuthContext>
          <HeaderBar
            loggedIn={user.signedIn}
            innerNav={compId != 0}
            compIdChanged={setCompId}
          />
          <Layout>
            <PageRouter
              authStateReceived={authStateReceived}
              signedIn={user.signedIn}
              compId={compId}
              loading={loading}
              competitionInfo={competitionInfo}
              updateCompId={setCompId}
            />
          </Layout>
        </AuthContext>

Reply:
Makes sense, nice one :)